### PR TITLE
Link balance_updates to operations

### LINKED ISSUE #471: Where possible the balance_updates table should include operation IDs
There is no user-friendly way to link `balance_updates` records to specific operations. Where possible, the table should include a value for an `operation_group_hash` column. 

This should apply to records whose source is either `operation` or `operation_result`.

### PR DESCRIPTION
**Pull request by ba205**
resolves #471

### CODE REVIEW

**Comment by anonymoussprocket** [comment]
@ba205, the build is failing, please review

**Comment by ivanopagano** [comment]
@vishakh @ba205 @anonymoussprocket please consider my note [here](https://github.com/Cryptonomic/Conseil/issues/471#issuecomment-525271046)

File: doc/conseil.sql [hunk 1]
```diff
@@ -80,7 +80,8 @@ CREATE TABLE public.balance_updates (
     change numeric NOT NULL,
     level numeric,
     delegate character varying,
-    category character varying
+    category character varying,
+    operation_group_hash character varying
```
Comment (ivanopagano) [inline doc/conseil.sql]: Is this actually needed? see [this note](https://github.com/Cryptonomic/Conseil/issues/471#issuecomment-525271046)

File: src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala [hunk 2]
```diff
@@ -225,7 +225,7 @@ class TezosDatabaseOperationsTest
             val generatedUpdateRows =
             operation
               .convertToA[List, Tables.BalanceUpdatesRow]
-              .map(_.copy(sourceId = Some(opRow.operationId)))
+              .map(_.copy(sourceId = Some(opRow.operationId), operationGroupHash = Some(opRow.operationGroupHash)))
```
Comment (ivanopagano) [inline src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala]: Needed?

File: doc/conseil.sql (hunk 1 above)
Comment (ba205) [reply doc/conseil.sql]: So the issue is I wanted a quick fix, and as far as I saw in the table in dev, source_hash only works for sources that are blocks. When I tried to do a similar approach to place operation hashes in source_hash, I ended up somehow not capturing block sources, so I chose the easier way out this time.

File: src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala (hunk 2 above)
Comment (ba205) [reply src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala]: See above. This is the quick fix that works, in running Lorre and in tests

**Comment by ba205** [comment]
> I need a feedback regarding the necessity to reference the operation group from balance updates.
> 
> Additionally: is it correct that we don't test the platform discovery attributes for balance updates? Do we expose that table to the rest client?

@piotrkosecki @anonymoussprocket Do you guys know how balance_updates is exposed?

**Comment by ba205** [comment]
@anonymoussprocket @piotrkosecki @ivanopagano I will make a new pr, as I meant to merge this branch into another, not directly into master. I'll keep this open until questions have been resolved however.

File: doc/conseil.sql (hunk 1 above)
Comment (ivanopagano) [reply doc/conseil.sql]: I'm not sure if I'm missing anything, but I was trying to point out that, for balance updates referring to "operation" or "operation_result" in the `balance_updates.source` column, the referred operation can be found by reading the `operation_id` written in the `balance_updates.source_id` column.

I see no need of the operation group hash, you can simply get the operations row and read the operation group hash from that...

Comment (ba205) [reply doc/conseil.sql]: True, but we're leaning towards a more normalized data model with as few joins as possible, iirc. @anonymoussprocket thoughts?

Comment (ba205) [reply doc/conseil.sql]: I'm pretty sure this has to do with ease of use for Arronax specifically. Correct me if I'm wrong.

**Comment by ba205** [comment]
> @anonymoussprocket @piotrkosecki @ivanopagano I will make a new pr, as I meant to merge this branch into another, not directly into master. I'll keep this open until questions have been resolved however.

Update, I didn't realize the other branch was already merged and deleted, this branch is fine

**Comment by piotrkosecki** [comment]
> > I need a feedback regarding the necessity to reference the operation group from balance updates.
> > Additionally: is it correct that we don't test the platform discovery attributes for balance updates? Do we expose that table to the rest client?
> 
> @piotrkosecki @anonymoussprocket Do you guys know how balance_updates is exposed?

it should be exposed in the rest API in config files: https://github.com/Cryptonomic/Conseil/blob/master/src/main/resources/metadata/tezos.alphanet.conf
https://github.com/Cryptonomic/Conseil/blob/master/src/main/resources/metadata/tezos.mainnet.conf
if it's not mentioned in those config files it won't be visible through API

**Pull request closed without merging**